extension/chrome/elements/composer/composer-input.ts

```diff
@@ -43,23 +59,66 @@ export class ComposerInput extends ComposerComponent {
 
   // -- private
 
-  private textPastedIntoBodyHandler = (clipboardEvent: ClipboardEvent) => {
-    if (!clipboardEvent.clipboardData) {
-      return;
-    }
-    const clipboardHtmlData = clipboardEvent.clipboardData.getData('text/html');
-    if (!clipboardHtmlData) {
-      return; // if it's text, let the original handlers paste it
-    }
-    clipboardEvent.preventDefault();
-    clipboardEvent.stopPropagation();
-    const sanitized = Xss.htmlSanitizeAndStripAllTags(clipboardHtmlData, '<br>');
-    // the lines below simulate ctrl+v, but not perfectly (old selected text does not get deleted)
-    const selection = window.getSelection();
-    if (selection) {
-      const r = selection.getRangeAt(0);
-      r.insertNode(r.createContextualFragment(sanitized));
-    }
+  private handlePaste = () => {
+    this.squire.addEventListener('willPaste', (e: WillPasteEvent) => {
+      const plainTextDiv = document.createElement('div');
+      plainTextDiv.appendChild(e.fragment);
+      plainTextDiv.innerHTML = this.richTextMode ? Xss.htmlSanitize(plainTextDiv.innerHTML) : Xss.htmlSanitizeAndStripAllTags(plainTextDiv.innerHTML, '<br>'); // xss-sanitized
+      e.fragment.appendChild(plainTextDiv);
+    });
+  }
+
+  private handlePasteImages = () => {
+    this.squire.addEventListener('drop', (e: DragEvent) => {
```

Comment:
I can't make it work because of types:

```
Argument of type '(el: HTMLElement, e: DragEvent) => void' is not assignable to parameter of type '(e: HTMLElement, event: Event<HTMLElement, null>) => void | Promise<void>'
```

@tomholub Could you please help me with this one?

Reply:
Since it's not exactly a common case (normally we use jquery), we can just give it a try/catch like below.

Reply:
resolved